webapp/src/shared/context/NotificationContext.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+
+import { SnackBar, SnackbarType, SnackbarPosition } from '../components';
+
+const NOTIFICATION_SPAN_MS = 4000;
+
+type Context = {
+  notify: (message: string) => void;
+  warn: (message: string) => void;
+  hide: () => void;
+};
+
+type NotificationContextProps = {
+  children: React.ReactNode;
+};
+
+type Notification = {
+  message: string;
+  type: SnackbarType;
+  position: SnackbarPosition;
+};
+
+const NotificationContext = React.createContext<Context | undefined>(undefined);
+
+export const NotificationContextProvider = ({
+  children,
+}: NotificationContextProps) => {
+  const [notification, setNotification] = React.useState<Notification>();
+  const [visible, setVisible] = React.useState<boolean>(false);
+
+  const notify = (message: string) => {
+    if (message.length) {
+      setNotification({ message, type: 'info', position: 'top' });
+      setVisible(true);
+    }
+  };
+
+  const warn = (message: string) => {
+    if (message.length) {
+      setNotification({ message, type: 'warning', position: 'bottom' });
+      setVisible(true);
+    }
```

Comment:
Since this was new and didn't have a design for this I kind of made this up:
- informative notifications -> will display in the top of the page with green color (like the text we had)
- warning notifications -> will display in the bottom of the page with yellow color (like the text we had)

I'm not particularly in love with this decision, we could have both appear from bottom and use top notifications for new future types of notifications we want to display. Or maybe use another colors, I don't mind.